src/Shared/ProblemDetails/HttpValidationProblemDetailsJsonConverter.cs

```diff
@@ -30,7 +34,8 @@ public static HttpValidationProblemDetails ReadProblemDetails(ref Utf8JsonReader
         {
             if (reader.ValueTextEquals(Errors.EncodedUtf8Bytes))
             {
-                var errors = DeserializeErrors(ref reader, options);
+                var context = new ErrorsJsonContext(options);
```

Comment:
I just looked at the build errors. Considering this mutates the JsonSerializerOptions, should we be configuring this earlier at startup. I know @brunolins16 has had to consider similar issues working on #45405.

Reply:
I understand that using an internal `JsonContext` will make it easier to users, since they won't need to be annoyed with a bunch of types that are used internally, however, since those converters are internal is not even possible to use (HttpValidation)ProblemDetails with JSON source gen https://github.com/dotnet/aspnetcore/issues/43236 I believe we should think more about how to make `ProblemDetails` serialization safer, in special those converters that in my opinion are not really needed and are the source of the most of the issues.

Reply:
I believe if the converters are not there, JSON would be able to source gen a `JsonTypeInfo<IDictionary<string, string[]>` only adding `HttpValidationProblemDetails` to the application `JsonContext`

Reply:
I have a proposal here https://github.com/dotnet/aspnetcore/issues/44132#issuecomment-1256683443

Reply:
Working on this in https://github.com/dotnet/aspnetcore/pull/45646